### PR TITLE
Move workaround `utils._getitem`/`_setitem` to `chainerx`

### PR DESCRIPTION
**Pull request by niboshi**

### CODE REVIEW

**Comment by hvy** [comment]
Is `_fallback_workarounds` missing?

**Comment by niboshi** [comment]
Oops, added.

File: chainerx/_fallback_workarounds.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                array,
+                array.device.index),
+            0),
+        strides=array.strides)
+
+
+def _from_cupy(array):
+    if cupy is None:
+        raise RuntimeError(
+            'Currently cupy is required in this operation.')
```
Comment (hvy) [inline chainerx/_fallback_workarounds.py]: A detail but you can assert that `cupy is not None`. Maybe this error should be thrown by the caller. That would be friendlier than "ChainerX fallback implementation only supports native or cuda backends.", in case cupy isn't available.

File: chainerx/_docs/__init__.py [hunk 2]
```diff
@@ -10,7 +10,8 @@
 
 
 def set_doc(obj, docstring):
-    if inspect.ismethoddescriptor(obj) or inspect.isroutine(obj):
+    if ((inspect.ismethoddescriptor(obj) or inspect.isroutine(obj))
+            and not inspect.isfunction(obj)):
```
Comment (hvy) [inline chainerx/_docs/__init__.py]: Sorry but could you explain why this change is needed?

File: chainerx/_fallback_workarounds.py (hunk 1 above)
Comment (hvy) [reply chainerx/_fallback_workarounds.py]: Okay, let's leave it here as is if it was meant to be reused. It is currently only called from `_to_chainerx` (which I mean by "caller", sorry if it was unclear) and this code is never executed so that's why I asked.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
Could you check the Travis failure?

File: chainerx/_fallback_workarounds.py (hunk 1 above)
Comment (niboshi) [reply chainerx/_fallback_workarounds.py]: Ah, I see. Then I think that's better.

Comment (niboshi) [reply chainerx/_fallback_workarounds.py]: Fixed as suggested.

**Comment by niboshi** [comment]
Fixed.

File: chainerx/_fallback_workarounds.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    return chainerx.to_numpy(array, copy=False)
+
+
+def _from_numpy(array):
+    assert isinstance(array, numpy.ndarray)
+    return chainerx.array(array, copy=False)
+
+
+def _to_cupy(array):
+    assert cupy is not None
```
Comment (hvy) [inline chainerx/_fallback_workarounds.py]: I think this assertion may fail (although we can assume it in `_from_cupy`), i.e. ChainerX with CUDA backend is used but CuPy is unavailable?

Comment (hvy) [reply chainerx/_fallback_workarounds.py]: I must have missed the check prior to the call. This doesn't seem like an issue.

**Comment by hvy** [comment]
LGTM!

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
jenkins, test this please.
